### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by LastValentina**

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1 @@
+[1210/174755.882:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (shoggy) [inline debug.log:1]: Add *.log files to `.gitignore` please

File: pom.xml [hunk 2]
```diff
@@ -61,6 +70,11 @@
             <version>${testng.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>selenium-java</artifactId>
+            <version>3.141.59</version>
```
Comment (shoggy) [inline pom.xml]: Move version to `properties` section and add `<scope>test` please

File: src/test/java/hw2/ex1/Ex1Test.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    String pass = "Jdi1234";
+    String user = "ROMAN IOVLEV";
+    String[] header = {"HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS"};
+    String[] left = {"Home", "Contact form", "Service", "Metals & Colors", "Elements packs"};
+    String[] keyWord = new String[4];
+
+    @BeforeClass
+    public void setUp() {
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
```
Comment (shoggy) [inline src/test/java/hw2/ex1/Ex1Test.java]: Let's try to start the browser already maximized. Use `ChromeOptions`

File: src/test/java/hw2/ex1/Ex1Test.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        driver.manage().timeouts().implicitlyWait(1, TimeUnit.SECONDS);
+        driver.manage().timeouts().pageLoadTimeout(3, TimeUnit.SECONDS);
```
Comment (shoggy) [inline src/test/java/hw2/ex1/Ex1Test.java]: What is the reason for overriding default values?

File: src/test/java/hw2/ex1/Ex1Test.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.concurrent.TimeUnit;
+
+public class Ex1Test {
+    WebDriver driver;
+    String url = "https://jdi-testing.github.io/jdi-light/index.html";
+    String login = "Roman";
+    String pass = "Jdi1234";
+    String user = "ROMAN IOVLEV";
+    String[] header = {"HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS"};
+    String[] left = {"Home", "Contact form", "Service", "Metals & Colors", "Elements packs"};
```
Comment (shoggy) [inline src/test/java/hw2/ex1/Ex1Test.java]: can be `final` or `static final`

File: src/test/java/hw2/ex1/Ex1Test.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            String actualText = lsHeader.get(i).getText();
+            softAssert.assertEquals(actualText, header[i]);  //5.1# check 1-3 items of header
+        }
+        String actText = driver.findElement(By.cssSelector("ul.uui-navigation.nav.navbar-nav.m-l8 [href='metals-colors.html']"))
+                .getText();
+        softAssert.assertEquals(actText, header[3]);  //5.2# check 4th items of header
+
+        //6. four images on Index page are displayed
+        List<WebElement> listBenefitIcons = driver.findElements
+                (By.cssSelector("div.row.clerafix.benefits span[class~=icons-benefit]"));
```
Comment (shoggy) [inline src/test/java/hw2/ex1/Ex1Test.java]: in css classes can be referenced in any order:  a.first.second == a.second.first.
```suggestion
                (By.cssSelector("div.row.clerafix.benefits span.icons-benefit"));
```

File: src/test/java/hw2/ex1/Ex1Test.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        List<WebElement> listFButton = driver.findElements(By.cssSelector("[value='Frame Button']"));
+        softAssert.assertEquals(listFButton.size(), 1);
+
+        //10.Switch to original window back
+        driver.switchTo().defaultContent();
+
+        //11.Assert that there are 5 items in the Left Section are displayed and they have proper text
+        WebElement elSideberMenu = driver.findElement(By.cssSelector("ul.sidebar-menu"));
+        for (String s : left) {
+            String locatorSearchElementLeftSection = "//span[text()=" + "\'" + s + "\'" + "]";
```
Comment (shoggy) [inline src/test/java/hw2/ex1/Ex1Test.java]: `"\'"` 
If you want single quote, it does not require escape
If you want backslash then it should be `\\`

File: src/test/java/hw2/ex2/Ex2Test.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        softAssert.assertEquals(driver.getCurrentUrl(), url);
+
+        //2. Browser title equals "Home Page"
+        softAssert.assertEquals(driver.getTitle(), "Home Page");
+
+        //3. Perform login
+        driver.findElement(By.id("user-icon")).click();
+        driver.findElement(By.id("name")).sendKeys(login);
+        driver.findElement(By.id("password")).sendKeys(pass);
+        driver.findElement(By.id("login-button")).click();
```
Comment (shoggy) [inline src/test/java/hw2/ex2/Ex2Test.java]: looks the same as in Ex1Test. If so, then common parts can be moved to the parent class

File: src/test/java/hw2/ex2/Ex2Test.java [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        for (int i = 0; i < elementsTest.length; i++) {
+            int counterLog = listLog.size() - i - 1;
+            boolean actualContElement = listLog.get(counterLog).getText().contains(elementsTest[i][0]);
+            boolean actualContent;
+            if (elementsTest[i][1].equals("cbox")) {
+                actualContent = listLog.get(counterLog).getText().contains("condition changed to true");
+            } else {
+                actualContent = listLog.get(counterLog).getText().contains("value changed to");
+            }
+            softAssert.assertTrue(actualContElement & actualContent, "Log for defined element doesn't relevanl");
```
Comment (shoggy) [inline src/test/java/hw2/ex2/Ex2Test.java]: for boolean expressions `&` evaluates both operands, `&&` uses short-circuit evaluation. Be careful

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/test/java/hw2/ex2/Ex2Test.java (hunk 8 above)
Comment (LastValentina) [reply src/test/java/hw2/ex2/Ex2Test.java]: Parent abstract class LoginTest was added

File: debug.log (hunk 1 above)
Comment (LastValentina) [reply debug.log:1]: done

File: src/test/java/hw2/ex1/Ex1Test.java (hunk 4 above)
Comment (LastValentina) [reply src/test/java/hw2/ex1/Ex1Test.java]: implicitly wait was changed to explicitly (methods for them defined in class LoginTest (package hw2).

**Review by shoggy — APPROVED** [review]

**Pull request merged**